packages/docusaurus/src/server/plugins/index.ts

```diff
@@ -104,22 +103,15 @@ export async function loadPlugins(context: LoadContext): Promise<{
         plugin.name,
         pluginId,
       );
-      // TODO this would be better to do all that in the codegen phase
-      // TODO handle context for nested routes
```

Comment:
It works everywhere, because it's handled in the same way as route components are handled. As long as there are `RouteContextProvider`s, the nested routes will be properly merged. Unless "nested routes" mean something different here?

Reply:
Hmm, I think it "works" yes, but only parent routes are namespacing the context data under a "data" attribute while nested routes expose the context data

Maybe if we had a good case to test this feature on docs route we'll see if it works?

This looks fine for now, we'll fix edge cases as soon as we have a use-case for nested context